src/ctia/schemas/coa.clj

```diff
@@ -10,56 +10,28 @@
 (s/defschema COA
   (merge
    c/GenericStixIdentifiers
-   {:valid_time c/ValidTime
-    (s/optional-key :stage)
-
-    (describe v/COAStage
-              "specifies what stage in the cyber threat management lifecycle this Course Of Action is relevant to ")
-
-    (s/optional-key :coa_type)
-    (describe v/COAType
-              "type of this CourseOfAction")
-
-    (s/optional-key :objective)
-    (describe [s/Str]
-              "characterizes the objective of this Course Of Action") ;; Squashed / simplified
-
-    (s/optional-key :impact)
-    (describe s/Str
-              "characterizes the estimated impact of applying this Course Of Action")
-
-    (s/optional-key :cost)
-    (describe v/HighMedLow
-              "characterizes the estimated cost for applying this Course Of Action")
-
-    (s/optional-key :efficacy)
-    (describe v/HighMedLow
-              "effectiveness of this Course Of Action in achieving its targeted Objective")
-
-    (s/optional-key :source)
-    (describe s/Str
-              "Source of this Course Of Action")
-
-    (s/optional-key :related_COAs)
-    (describe rel/RelatedCOAs
-              "identifies or characterizes relationships to one or more related courses of action")
-
-    ;; Not provided: handling
-    ;; Not provided: parameter_observables ;; Technical params using the CybOX language
-    ;; Not provided: structured_COA ;; actionable structured representation for automation
-    }))
-
-(s/defschema Type
-  (s/enum "COA"))
+   {:valid_time c/ValidTime}
+   (st/optional-keys
+    {:stage (describe v/COAStage "specifies what stage in the cyber threat management lifecycle this Course Of Action is relevant to")
+     :coa_type (describe v/COAType "type of this CourseOfAction")
+     :objective (describe [s/Str] "characterizes the objective of this Course Of Action") ;; Squashed / simplified
+     :impact (describe s/Str "characterizes the estimated impact of applying this Course Of Action")
+     :cost (describe v/HighMedLow "characterizes the estimated cost for applying this Course Of Action")
+     :efficacy (describe v/HighMedLow "effectiveness of this Course Of Action in achieving its targeted Objective")
+     :source (describe s/Str "Source of this Course Of Action")
+     :related_COAs (describe rel/RelatedCOAs "identifies or characterizes relationships to one or more related courses of action")
+     ;; Not provided: handling
+     ;; Not provided: parameter_observables ;; Technical params using the CybOX language
+     ;; Not provided: structured_COA ;; actionable structured representation for automation
+     })))
```

Comment:
Keys and value schemas are equally important to me.  Schemas look great without the descriptions...

```clojure
{:foo FooSchema}
```

Since we need the descriptions, I added the whitespace as a compromise to improve readability.  Maybe something else could improve on the white space, but it was a decent first attempt.  Readability should be our #2 goal (#1 being "it works").